src/proto/FunctionRpc.proto

```diff
@@ -262,6 +267,51 @@ message RpcFunctionMetadata {
   bool is_proxy = 7;
 }
 
+// Host tells worker it is ready to receive metadata
+message WorkerFunctionMetadataRequest {
+  // base directory for function app
+  string directory = 1;
+}
+
+// Worker sends function metadata back to host
+message WorkerFunctionMetadataResponse {
+  // list of function indexing responses
+  repeated WorkerFunctionMetadata results = 1;
+
+  // status of overall metadata request
+  StatusResult overall_status = 2;
+}
+
+// Message type that holds function metadata information
+message WorkerFunctionMetadata {
+  // function name
+  string name = 4;
+
+  // base directory for the Function
+  string directory = 1;
+
+  // Script file specified
+  string script_file = 2;
+
+  // Entry point specified
+  string entry_point = 3;
+
+  // Function id
+  string id = 11;
+
+  // Bindings info
+  repeated string bindings = 6;
+
+  // Is set to true for proxy
+  bool is_proxy = 7;
+
+  // Function indexing status
+  StatusResult status = 8;
```

Comment:
We should eliminate sending status here  -- worker should only send function metadata requests for valid functions ?